File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        // same as lower bound
        int low = 0, high = nums.length-1;
        int ind  = -1;
        int val = Integer.MIN_VALUE;
        while(low <= high){
            int mid = (low + high) / 2;
            if(nums[mid] == target) return mid;
            if(nums[mid] < target && val < nums[mid]){
                ind = mid;
                val = nums[mid];
            }else if(nums[mid] < target){
                low = mid + 1;
            }else{
                high = mid - 1;
            }
        }
        return ind+1;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.c
int searchInsert(int* nums, int n, int target) {
    // int i;
    // for(i = 0;i < n;i++){
    //     if(nums[i] >= target){
    //         return i;
    //     }
    // }
    // return n;


    int low = 0,high = n-1;
    while(low <= high){
        int mid = low + (high - low) / 2;
        if(nums[mid] == target) return mid;
        if(nums[mid] < target){
            low = mid + 1;
        }else{
            high = mid - 1;
        }
    }
    return low;
}